back-end/routes/clusters.js

```diff
@@ -0,0 +1,208 @@
+const express = require('express');
+const { PrismaClient } = require('@prisma/client');
+const fetch = require('node-fetch');
+const router = express.Router();
+const prisma = new PrismaClient();
+
+router.use(express.json());
+
+const fetchAudioFeatures = async (songs, accessToken) => {
+    const songIds = songs.map(song => (song.uri).split(':').pop());
+    const stringOfIds = songIds.join(',');
+    try {
+        const response = await fetch(`https://api.spotify.com/v1/audio-features?ids=${encodeURIComponent(stringOfIds)}`, {
+            headers: {
+                'Authorization': `Bearer ${accessToken}`
+            }
+        });
+        if (!response.ok) {
+            throw new Error(`HTTP error! status: ${response.status}`);
+        }
+        const audioData = await response.json();
+        return audioData.audio_features;
+    } catch (error) {
+        console.error(error);
+        return [];
+    }
+};
+
```

Comment:
Didn't we have this function elsewhere?

Reply:
Yes, we have the functions in the recommendation. I will make a new file for those commonly used, global methods so they can then be used for other uses like we discussed in the 1:1 today.